SSH Tunnel Fails with "The connection attempt failed." despite successful SSH login (Metabase Cloud)

**Describe the bug**

This indicates the issue is not with the database or tunnel, but with how **Metabase Cloud handles SSH connections**.

### Steps to Reproduce

1. Configure a PostgreSQL database with SSH tunnel access in Metabase Cloud.
2. Use a valid SSH key and correct database credentials.
3. Test the connection.
4. Observe the failure with the error above.

### Expected Behavior

Metabase Cloud should successfully connect to the database via SSH as the local Metabase instance does.

### Notes

The same setup works locally with an older Metabase version, confirming that the issue is isolated to the Cloud environment.

tested with the `:latest` docker image `https://hub.docker.com/layers/metabase/metabase/latest/images/sha256-c42cc28aa2aec16283118ac60f975e3c588973b8416d0a2222f8cbad5020ce59` against an older version older version `metabase/metabase:v0.52.15.2`

the older version `metabase/metabase:v0.52.15.2` didn't have the issue, and the connection was successful


however the `:latest` image failed to connect and here is the log of the 
```
{:status "initializing", :progress 0.95}

2025-04-04 04:20:35,917 INFO task.refresh-slack-channel-user-cache :: Slack is not configured, not refreshing slack user/channel cache.
2025-04-04 04:20:36,348 DEBUG middleware.log :: GET /api/health 200 1ms (0 DB calls) App DB connections: 0/4 Jetty threads: 5/50 (4 idle, 0 queued) (54 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2025-04-04 04:20:36,712 INFO i18n.impl :: Reading available locales from locales.clj...
2025-04-04 04:20:36,958 INFO util.fonts :: Reading available fonts from /frontend_client/app/fonts
2025-04-04 04:20:38,204 DEBUG middleware.log :: GET /api/user/current 401 26ms (0 DB calls) {:metabase-user-id nil} 
"Unauthenticated"

2025-04-04 04:20:38,594 DEBUG middleware.log :: GET /api/session/properties 200 462ms (2 DB calls) App DB connections: 0/4 Jetty threads: 5/50 (2 idle, 0 queued) (55 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2025-04-04 04:20:40,606 DEBUG middleware.log :: POST /api/util/password_check 200 39ms (0 DB calls) App DB connections: 0/4 Jetty threads: 5/50 (2 idle, 0 queued) (55 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2025-04-04 04:20:43,347 INFO jdbcjobstore.JobStoreTX :: ClusterManager: detected 2 failed or restarted instances.
2025-04-04 04:20:43,348 INFO jdbcjobstore.JobStoreTX :: ClusterManager: Scanning for instance "d2538c0a57e11743740422719"'s failed in-progress jobs.
2025-04-04 04:20:43,352 INFO jdbcjobstore.JobStoreTX :: ClusterManager: Scanning for instance "d2538c0a57e11743740426414"'s failed in-progress jobs.
2025-04-04 04:20:51,154 INFO models.user :: Setting User 1's last_acknowledged_version to v0.53.9, the current version
2025-04-04 04:20:51,195 INFO models.user :: Adding User 1 to All Users permissions group...
2025-04-04 04:20:51,195 INFO models.user :: Adding User 1 to All Users permissions group...
2025-04-04 04:20:51,413 DEBUG middleware.log :: POST /api/setup 200 474ms (51 DB calls) App DB connections: 0/4 Jetty threads: 5/50 (2 idle, 0 queued) (56 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2025-04-04 04:20:51,570 DEBUG middleware.log :: GET /api/session/properties 200 135ms (20 DB calls) App DB connections: 0/4 Jetty threads: 6/50 (2 idle, 0 queued) (56 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
2025-04-04 04:20:51,610 DEBUG middleware.log :: GET /api/setting 200 176ms (16 DB calls) App DB connections: 0/4 Jetty threads: 6/50 (2 idle, 0 queued) (56 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
2025-04-04 04:21:24,210 INFO driver.impl :: Initializing driver :postgres...
2025-04-04 04:21:24,814 WARN keyverifier.AcceptAllServerKeyVerifier :: Server at /<<DEDUCTED>>:22 presented unverified EC key: <<DEDUCTED>>
2025-04-04 04:21:25,897 WARN keyverifier.AcceptAllServerKeyVerifier :: Server at /<<DEDUCTED>>:22 presented unverified EC key: <<DEDUCTED>>
2025-04-04 04:21:26,699 WARN forward.DefaultForwarder :: Failed (SshChannelOpenException) to open channel for session=Nio2Session[local=/127.0.0.1:38247, remote=/127.0.0.1:32950]: Connection refused
2025-04-04 04:21:26,701 WARN forward.DefaultForwarder :: exceptionCaught(Nio2Session[local=/127.0.0.1:38247, remote=/127.0.0.1:32950]) NullPointerException: Cannot invoke "org.apache.sshd.common.io.IoOutputStream.writeBuffer(org.apache.sshd.common.util.buffer.Buffer)" because the return value of "org.apache.sshd.common.forward.TcpipClientChannel.getAsyncIn()" is null
java.lang.NullPointerException: Cannot invoke "org.apache.sshd.common.io.IoOutputStream.writeBuffer(org.apache.sshd.common.util.buffer.Buffer)" because the return value of "org.apache.sshd.common.forward.TcpipClientChannel.getAsyncIn()" is null
	at org.apache.sshd.common.forward.DefaultForwarder$StaticIoHandler.lambda$messageReceived$2(DefaultForwarder.java:1063)
	at org.apache.sshd.common.util.threads.ThreadUtils.runAsInternal(ThreadUtils.java:68)
	at org.apache.sshd.common.forward.DefaultForwarder$StaticIoHandler.messageReceived(DefaultForwarder.java:1063)
	at org.apache.sshd.common.io.nio2.Nio2Session.handleReadCycleCompletion(Nio2Session.java:409)
	at org.apache.sshd.common.io.nio2.Nio2Session$1.onCompleted(Nio2Session.java:382)
	at org.apache.sshd.common.io.nio2.Nio2Session$1.onCompleted(Nio2Session.java:377)
	at org.apache.sshd.common.io.nio2.Nio2CompletionHandler.lambda$completed$0(Nio2CompletionHandler.java:38)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.apache.sshd.common.io.nio2.Nio2CompletionHandler.completed(Nio2CompletionHandler.java:37)
	at java.base/sun.nio.ch.Invoker.invokeUnchecked(Unknown Source)
	at java.base/sun.nio.ch.Invoker$2.run(Unknown Source)
	at java.base/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
2025-04-04 04:21:26,717 ERROR driver.util :: Failed to connect to Database
org.postgresql.util.PSQLException: The connection attempt failed.
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:364)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:54)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:273)
	at org.postgresql.Driver.makeConnection(Driver.java:446)
	at org.postgresql.Driver.connect(Driver.java:298)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at clojure.java.jdbc$get_driver_connection.invokeStatic(jdbc.clj:271)
	at clojure.java.jdbc$get_driver_connection.invoke(jdbc.clj:250)
	at clojure.java.jdbc$get_connection.invokeStatic(jdbc.clj:411)
	at clojure.java.jdbc$get_connection.invoke(jdbc.clj:274)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1111)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invokeStatic(connection.clj:398)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invoke(connection.clj:395)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_$fn__64205.invoke(connection.clj:407)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection$fn__64187.invoke(connection.clj:383)
	at metabase.util.ssh$do_with_ssh_tunnel.invokeStatic(ssh.clj:161)
	at metabase.util.ssh$do_with_ssh_tunnel.invoke(ssh.clj:154)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invokeStatic(connection.clj:378)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invoke(connection.clj:371)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invokeStatic(connection.clj:406)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invoke(connection.clj:402)
	at metabase.driver.sql_jdbc$fn__118054.invokeStatic(sql_jdbc.clj:50)
	at metabase.driver.sql_jdbc$fn__118054.invoke(sql_jdbc.clj:48)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__55450.invoke(util.clj:173)
	at clojure.core$binding_conveyor_fn$fn__5842.invoke(core.clj:2047)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.EOFException
	at org.postgresql.core.PGStream.receiveChar(PGStream.java:478)
	at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:607)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:201)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:268)
	... 36 more
2025-04-04 04:21:26,720 ERROR api.database :: Cannot connect to Database
clojure.lang.ExceptionInfo: The connection attempt failed. {:message "The connection attempt failed."}
	at metabase.driver.util$can_connect_with_details_QMARK_.invokeStatic(util.clj:190)
	at metabase.driver.util$can_connect_with_details_QMARK_.doInvoke(util.clj:161)
	at clojure.lang.RestFn.invoke(RestFn.java:445)
	at metabase.api.database$test_database_connection.invokeStatic(database.clj:731)
	at metabase.api.database$test_database_connection.doInvoke(database.clj:721)
	at clojure.lang.RestFn.invoke(RestFn.java:428)
	at metabase.api.database$test_connection_details106129__106130.invokeStatic(database.clj:785)
	at metabase.api.database$test_connection_details106129__106130.invoke(database.clj:765)
	at metabase.api.database$fn__106137$fn__106143.invoke(database.clj:809)
	at metabase.api.macros$_core_fn92811__92812$core_fn__92813.invoke(macros.clj:405)
	at metabase.api.macros$endpoint_handler_STAR_92855__92857$handler__92861.invoke(macros.clj:601)
	at metabase.api.macros$build_ns_handler92888__92889$ns_handler_STAR___92890.invoke(macros.clj:665)
	at metabase.api.open_api.HandlerWithOpenAPISpec.invoke(open_api.clj:42)
	at metabase.server.middleware.auth$enforce_authentication$fn__62465.invoke(auth.clj:23)
	at metabase.api.open_api.HandlerWithOpenAPISpec.invoke(open_api.clj:42)
	at metabase.api.util.handlers$_route_map_handler$fn__102683.invoke(handlers.clj:23)
	at metabase.api.open_api.HandlerWithOpenAPISpec.invoke(open_api.clj:42)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620$f__102621$respond_SINGLEQUOTE___102622.invoke(core.clj:197)
	at metabase.api.routes$fn__111125.invokeStatic(routes.clj:97)
	at metabase.api.routes$fn__111125.invoke(routes.clj:96)
	at metabase.api.open_api.HandlerWithOpenAPISpec.invoke(open_api.clj:42)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620.invoke(core.clj:200)
	at metabase.api.open_api.HandlerWithOpenAPISpec.invoke(open_api.clj:42)
	at metabase.server.routes$fn__111683$fn__111684.invoke(routes.clj:70)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620.invoke(core.clj:200)
	at compojure.core$make_context$handler__102648.invoke(core.clj:290)
	at compojure.core$make_context$fn__102652.invoke(core.clj:300)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620$f__102621$respond_SINGLEQUOTE___102622.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__102601.invoke(core.clj:153)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620$f__102621$respond_SINGLEQUOTE___102622.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__102601.invoke(core.clj:153)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620$f__102621$respond_SINGLEQUOTE___102622.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__102601.invoke(core.clj:153)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620$f__102621$respond_SINGLEQUOTE___102622.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__102601.invoke(core.clj:153)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620$f__102621$respond_SINGLEQUOTE___102622.invoke(core.clj:197)
	at metabase.api.util.handlers$_route_map_handler$fn__102683.invoke(handlers.clj:24)
	at metabase.api.open_api.HandlerWithOpenAPISpec.invoke(open_api.clj:42)
	at metabase.api.util.handlers$_route_map_handler$fn__102683.invoke(handlers.clj:23)
	at metabase.api.open_api.HandlerWithOpenAPISpec.invoke(open_api.clj:42)
	at compojure.core$routes$fn__102620$f__102621.invoke(core.clj:198)
	at compojure.core$routes$fn__102620.invoke(core.clj:200)
	at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__62642.invoke(exceptions.clj:111)
	at metabase.server.middleware.exceptions$catch_api_exceptions$fn__62639.invoke(exceptions.clj:100)
	at metabase.server.middleware.log$log_api_call$fn__112107$fn__112108$fn__112109.invoke(log.clj:232)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invokeStatic(diagnostic.clj:17)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invoke(diagnostic.clj:12)
	at metabase.server.middleware.log$log_api_call$fn__112107$fn__112108.invoke(log.clj:223)
	at toucan2.execute$do_with_call_counts.invokeStatic(execute.clj:112)
	at toucan2.execute$do_with_call_counts.invoke(execute.clj:103)
	at metabase.server.middleware.log$log_api_call$fn__112107.invoke(log.clj:222)
	at metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__117099.invoke(browser_cookie.clj:40)
	at metabase.server.middleware.security$add_security_headers$fn__62598.invoke(security.clj:273)
	at metabase.server.middleware.json$wrap_json_body$fn__48850.invoke(json.clj:128)
	at metabase.server.middleware.offset_paging$handle_paging$fn__116027.invoke(offset_paging.clj:48)
	at metabase.server.middleware.json$wrap_streamed_json_response$fn__48838.invoke(json.clj:83)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__117366.invoke(keyword_params.clj:55)
	at ring.middleware.params$wrap_params$fn__117385.invoke(params.clj:77)
	at metabase.server.middleware.misc$maybe_set_site_url$fn__111823.invoke(misc.clj:59)
	at metabase.server.middleware.session$reset_session_timeout$fn__117086.invoke(session.clj:270)
	at metabase.server.middleware.session$bind_current_user$fn__117079$fn__117080.invoke(session.clj:245)
	at metabase.request.session$do_with_current_user.invokeStatic(session.clj:47)
	at metabase.request.session$do_with_current_user.invoke(session.clj:30)
	at metabase.server.middleware.session$bind_current_user$fn__117079.invoke(session.clj:244)
	at metabase.server.middleware.session$wrap_current_user_info$fn__117071.invoke(session.clj:204)
	at metabase.analytics.sdk$embedding_mw$embedding_mw_fn__65418.invoke(sdk.clj:79)
	at metabase.server.middleware.session$wrap_session_id$fn__117043.invoke(session.clj:82)
	at metabase.server.middleware.auth$wrap_static_api_key$fn__62473.invoke(auth.clj:38)
	at ring.middleware.cookies$wrap_cookies$fn__117286.invoke(cookies.clj:200)
	at metabase.server.middleware.misc$add_content_type$fn__111805.invoke(misc.clj:28)
	at metabase.server.middleware.misc$disable_streaming_buffering$fn__111831.invoke(misc.clj:75)
	at ring.middleware.gzip$wrap_gzip$fn__117328.invoke(gzip.clj:86)
	at metabase.server.middleware.request_id$wrap_request_id$fn__111759.invoke(request_id.clj:10)
	at metabase.server.middleware.misc$bind_request$fn__111834$fn__111835.invoke(misc.clj:87)
	at metabase.request.current$do_with_current_request.invokeStatic(current.clj:19)
	at metabase.request.current$do_with_current_request.invoke(current.clj:15)
	at metabase.server.middleware.misc$bind_request$fn__111834.invoke(misc.clj:86)
	at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__117115.invoke(ssl.clj:41)
	at clojure.lang.Var.invoke(Var.java:395)
	at metabase.server.instance$async_proxy_handler$fn__62982.invoke(instance.clj:76)
	at metabase.server.instance.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:173)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.Server.handle(Server.java:563)
	at org.eclipse.jetty.server.HttpChannel$RequestDispatchable.dispatch(HttpChannel.java:1598)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:753)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:501)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:287)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:421)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:390)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:277)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:199)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:969)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1194)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1149)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.postgresql.util.PSQLException: The connection attempt failed.
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:364)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:54)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:273)
	at org.postgresql.Driver.makeConnection(Driver.java:446)
	at org.postgresql.Driver.connect(Driver.java:298)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at clojure.java.jdbc$get_driver_connection.invokeStatic(jdbc.clj:271)
	at clojure.java.jdbc$get_driver_connection.invoke(jdbc.clj:250)
	at clojure.java.jdbc$get_connection.invokeStatic(jdbc.clj:411)
	at clojure.java.jdbc$get_connection.invoke(jdbc.clj:274)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1111)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invokeStatic(connection.clj:398)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invoke(connection.clj:395)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_$fn__64205.invoke(connection.clj:407)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection$fn__64187.invoke(connection.clj:383)
	at metabase.util.ssh$do_with_ssh_tunnel.invokeStatic(ssh.clj:161)
	at metabase.util.ssh$do_with_ssh_tunnel.invoke(ssh.clj:154)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invokeStatic(connection.clj:378)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invoke(connection.clj:371)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invokeStatic(connection.clj:406)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invoke(connection.clj:402)
	at metabase.driver.sql_jdbc$fn__118054.invokeStatic(sql_jdbc.clj:50)
	at metabase.driver.sql_jdbc$fn__118054.invoke(sql_jdbc.clj:48)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__55450.invoke(util.clj:173)
	at clojure.core$binding_conveyor_fn$fn__5842.invoke(core.clj:2047)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	... 1 more
Caused by: java.io.EOFException
	at org.postgresql.core.PGStream.receiveChar(PGStream.java:478)
	at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:607)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:201)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:268)
	... 36 more
2025-04-04 04:21:26,739 DEBUG middleware.log :: POST /api/database 400 2558ms (1 DB calls) {:metabase-user-id 1} 
{:message "The connection attempt failed."}

```

--- a/src/metabase/driver/sql_jdbc/connection.clj
+++ b/src/metabase/driver/sql_jdbc/connection.clj
@@ -371,10 +371,7 @@ For setting the maximum, see [MB_APPLICATION_DB_MAX_CONNECTION_POOL_SIZE](#mb_ap
 (defn do-with-connection-spec-for-testing-connection
   "Impl for [[with-connection-spec-for-testing-connection]]."
   [driver details f]
-  (let [details (driver/incorporate-ssh-tunnel-details
-                  ;; If the tunnel is disabled this returned unchanged
-                 driver
-                 (update details :port #(or % (default-ssh-tunnel-target-port driver))))]
+  (let [details (update details :port #(or % (default-ssh-tunnel-target-port driver)))]
     (ssh/with-ssh-tunnel [details-with-tunnel details]
       (let [details-with-auth (driver.u/fetch-and-incorporate-auth-provider-details
                                driver

--- a/test/metabase/driver/sql_jdbc/connection_test.clj
+++ b/test/metabase/driver/sql_jdbc/connection_test.clj
@@ -372,8 +372,6 @@
                                      :tunnel-port ssh-test/ssh-mock-server-with-password-port
                                      :tunnel-user ssh-test/ssh-username
                                      :tunnel-pass ssh-test/ssh-password)]
-
-        (is (true? (driver.u/can-connect-with-details? (tx/driver) tunnel-db-details)))
         (mt/with-temp [:model/Database tunneled-db {:engine (tx/driver), :details tunnel-db-details}]
           (mt/with-db tunneled-db
             (sync/sync-database! (mt/db))

